### PR TITLE
pkg/query: Set different metadata on aggregated row to null

### PR DESCRIPTION
**Pull request by metalmatze**
Whenever we merge/aggregate functions we want to make sure that all meta data columns match with their content. 
For example, if the underlying filename is different, we have to set that column's value to `null`. 
This is to avoid showing misleading information.

### CODE REVIEW

**Comment by metalmatze** [comment]
FrostDB is updated in this PR and the replace in the go.mod is only to get https://github.com/apache/arrow/pull/37695 in. Once that merges we can simply remove that replace too.

File: pkg/query/flamegraph_arrow.go
Comment (brancz) [inline pkg/query/flamegraph_arrow.go:534]: all of these need to be wrapped in `if fb.builderX.IsValid`, and then an additional internal check that checks if the one being merged is null, as if that's the case it should also be set null. The reason both have to be checked is that setting something null after the fact can have the effect that the equality match might actually pass even if `.SetNull` was called, as `.SetNull` only updates the validity bitmap, not the value array.

Comment (metalmatze) [reply pkg/query/flamegraph_arrow.go:534]: Good catch! Updating.

File: pkg/query/flamegraph_arrow.go
Comment (brancz) [inline pkg/query/flamegraph_arrow.go:534]: I realize this seems drastic, but it has been shown that the negation here is a perf problem, so use `IsValid` rather than `IsNull`, they actually do different validity bitmap checks and therefore we won't loose on perf.

**Review by brancz — APPROVED** [review]
One tiny last thing, then lgtm

File: pkg/query/flamegraph_arrow.go
Comment (metalmatze) [reply pkg/query/flamegraph_arrow.go:534]: Right. I can add those later on. However, right now the underlying builders don't have those methods and I instead opted for `!IsNull`. Making this consistent with `IsValid` should be a follow-up.

Comment (brancz) [reply pkg/query/flamegraph_arrow.go:534]: Ah got it, ok lgtm then let's move forward

**Pull request merged**
